site_tools/utils_buildsize.py

```diff
@@ -138,15 +138,20 @@ def is_in_memory(name):
 
 
 def show_size(env, source, alias='__size'):
-    if env.has_key('CONFIG_DEVICE_MEMORY'):
-        action = Action(size_action, cmdstr="$SIZECOMSTR")
-    elif env.has_key('CONFIG_DEVICE_NAME'):
-        action = Action("$SIZE -C --mcu=$CONFIG_DEVICE_NAME %s" % source[0].path,
-                        cmdstr="$SIZECOMSTR")
-    else:
-        # use the raw output of the size tool
+    #this is a workaround for the riscv compilers
+    if 'riscv' in env['SIZE']:
         action = Action("$SIZE %s" % source[0].path,
                         cmdstr="$SIZECOMSTR")
+    else:
+        if env.has_key('CONFIG_DEVICE_MEMORY'):
+            action = Action(size_action, cmdstr="$SIZECOMSTR")
+        elif env.has_key('CONFIG_DEVICE_NAME'):
+            action = Action("$SIZE -C --mcu=$CONFIG_DEVICE_NAME %s" % source[0].path,
+                            cmdstr="$SIZECOMSTR")
```

Comment:
The `--mcu` arg is super AVR specific, and I actually don't quite like it. The action here should be switched by architecture somehow.